index.ts

```diff
@@ -0,0 +1,54 @@
+const prompt = require("prompt-sync")({ sigint: true });
+const upperValue: string | null = prompt("What value would you like FizzBuzz to go up to? ");
+console.log(`We go up to ${upperValue}`);
+
+function divisibleBy(num: number, divisor: number): boolean { return num % divisor == 0 };
+
+const key: {[index: number]: Function } = {
+    3: (answer: string[]) => {
+        answer.push('Fizz');
+        return answer;
+    },
+    5: (answer: string[]) => {
+        answer.push('Buzz');
+        return answer;
+    },
+    7: (answer: string[]) => {
+        if (answer.length < 2) {
+          answer.push('Bang');
+          return answer;
+        } else {
+          return ['Bang'];
+        } 
+    },
+    11: (answer: string[]) => answer = ['Bong'],
+    13: (answer: string[]) => {
+        let newAnswer: string[] = []
+        for (let i = 0; i < answer.length; i++) {
+            if (answer[i][0] == 'B') { newAnswer.push('Fezz') };
+            newAnswer.push(answer[i])
+        }
+        if (!newAnswer.includes('Fezz')) { newAnswer.push('Fezz') } 
+        return newAnswer;
+    },
+    17: (answer: string[]) => answer = answer.reverse()
+
+}
+
+const fizzBuzz = (upperValue: string | null) => {  
+  for (let n = 0; n <= Number(upperValue); n++) {
+    let answer: string[] = [];
+
+    for (const num of Object.keys(key)) {
+      if (divisibleBy(n, Number(num))) {
+        answer = key[Number(num)](answer)
+      }
+    }
+
+    if (answer.length == 0) { answer.push(String(n))}
```

Comment:
instead of using `String(n)` we could use` n.toString()`, I think this is more conventional